src/validation.cpp

```diff
@@ -2284,7 +2284,7 @@ bool CChainState::ConnectBlock(const CBlock& block, BlockValidationState& state,
 
     statsClient.timing("ConnectBlock_ms", (nTime8 - nTimeStart) / 1000, 1.0f);
     statsClient.gauge("blocks.tip.SizeBytes", ::GetSerializeSize(block, PROTOCOL_VERSION), 1.0f);
-    statsClient.gauge("blocks.tip.Height", m_chain.Height(), 1.0f);
+    statsClient.gauge("blocks.tip.Height", m_chain.Height() + 1, 1.0f); // without the +1, the "tip.Height" doesn't match rpc calls like `getblockcount`
```

Comment:
Good catch! Also, we call `ConnectBlock()` in a few more places which has nothing to do with the actual tip updates. I think all `blocks.tip` stats should be updated in `ConnectTip()` instead, in that case `m_chain.Height()` would work as expected. And it would probably make sense to update these stats in `DisconnectTip()` too for consistency sake.

Reply:
do you have a commit maybe? I don't fully understand

Reply:
smth like 446d19063434124c897ce1ed709cf92e580b1e4